### PR TITLE
Cleanup and refactor configurations

### PR DESCRIPTION
**Pull request by ramirezdiego**
The handling of some configurations is here slightly modified, to align with the format in the fuse configuration files (see https://github.com/XENONnT/private_nt_aux_files/pull/300).

In addition, the `help` field of some configs was extended for better documentation.

### CODE REVIEW

File: fuse/plugins/detector_physics/s2_photon_propagation.py [hunk 1]
```diff
@@ -317,19 +309,8 @@ def setup(self):
...
-            self.drift_velocity_scaling = 1.0
-            # calculating drift velocity scaling to match total drift time
-            # for R = 0 between cathode and gate
-            if "norm_drift_velocity" in self.enable_field_dependencies.keys():
-                if self.enable_field_dependencies["norm_drift_velocity"]:
-                    norm_dvel = self.field_dependencies_map_tmp(
-                        np.array([[0], [-self.tpc_length]]).T, map_name="drift_speed_map"
-                    )[0]
-                    norm_dvel *= 1e-4
-                    self.drift_velocity_scaling = self.drift_velocity_liquid / norm_dvel
```
Comment (ramirezdiego) [inline fuse/plugins/detector_physics/s2_photon_propagation.py:332]: @HenningSE Notice that I deleted this entire part, which I think was an accidental copypaste from the electron transport plugin.

**Comment by ramirezdiego** [comment]
Tests are currently failing because the fuse configuration file is not yet available in our database. The simulation should run by locally sourcing the file, as @cfuselli also verified.

File: docs/source/plugins/detector_physics/S2PhotonPropagation.rst [hunk 2]
```diff
@@ -141,10 +141,10 @@ S2PhotonPropagationBase plugin
      -
      - True
      - Width of the S2 area fraction top
-   * - enable_field_dependencies
+  * - enable_diffusion_transverse_map
```
Comment (mhliu0001) [inline docs/source/plugins/detector_physics/S2PhotonPropagation.rst:144]: Missing whitespace here?

Comment (ramirezdiego) [reply docs/source/plugins/detector_physics/S2PhotonPropagation.rst]: Fixed, thanks!

File: fuse/plugins/detector_physics/s2_photon_propagation.py [hunk 3]
```diff
@@ -583,7 +564,6 @@ def s2_pattern_map_diffuse(self, n_electron, z, xy, drift_time_mean):
             diffusion_constant_radial *= 1e-9  # cm²/ns
             diffusion_constant_azimuthal *= 1e-9  # cm²/ns
         else:
-            # diffusion_constant_transverse = diffusion_constant_transverse
             diffusion_constant_radial = self.diffusion_constant_transverse
             diffusion_constant_azimuthal = self.diffusion_constant_transverse
```
Comment (mhliu0001) [inline fuse/plugins/detector_physics/s2_photon_propagation.py:568]: Do we need a separate configuration for azimuthal diffusion constant?

Comment (mhliu0001) [reply fuse/plugins/detector_physics/s2_photon_propagation.py:568]: Also, `enable_diffusion_transverse_map` actually means whether to use a map for both transverse and azimuthal diffusion constant, so this name is a bit misleading for me.

Comment (ramirezdiego) [reply fuse/plugins/detector_physics/s2_photon_propagation.py:568]: When we do not use the map, radial and azimuthal both become the generic transverse diffusion coefficient, hence I would say this is correct.

Comment (ramirezdiego) [reply fuse/plugins/detector_physics/s2_photon_propagation.py:568]: Before it was tricky to make the difference, but I now added the `enable_` for the booleans that decide if we call the map or not.

**Comment by mhliu0001** [comment]
The error in the test is very strange. It was trying to look for `SIMULATION_CONFIG_FILE.json`, but it should be replaced by the real config file. Also, this occurs (only?) in `merge_cluster.py.`

**Comment by mhliu0001** [comment]
I guess we should update the microphysics context to include `SIMULATION_CONFIG_FILE.json` replacement.

File: fuse/plugins/micro_physics/merge_cluster.py [hunk 4]
```diff
@@ -62,7 +62,8 @@ class MergeCluster(FuseBasePlugin):
 
     # Config options
     tag_cluster_by = straxen.URLConfig(
-        default="energy",
+        default="take://resource://" "SIMULATION_CONFIG_FILE.json?&fmt=json" "&take=tag_cluster_by",
```
[the remark this answers is not in the document]
Comment (mhliu0001) [reply fuse/plugins/micro_physics/merge_cluster.py]: I pushed a fix myself to debug why the automatic test failed.

**Review by mhliu0001 — APPROVED** [review]
Now the tests have passed, so green light from my side.

File: docs/source/plugins/detector_physics/S2PhotonPropagation.rst
Comment (HenningSE) [inline docs/source/plugins/detector_physics/S2PhotonPropagation.rst:1]: We need to remove the drift_velocity_liquid config argument

**Review by HenningSE — CHANGES_REQUESTED** [review]
Looking good to me! We need to create a SR0 config with the same format to keep backwards compatibility. 

I will request changes here to keep track of this, if we have the config in place, we are good to merge.

**Pull request merged**